Makefile

```diff
@@ -59,8 +59,19 @@ lint:
 	@echo "checking lint"
 	@./.tool/lint
 
-test:
-	go test -v -race -cover $(shell go list ./... | grep -v /vendor/)
+.PHONY: .gofmt .gotest .govet
+
+PACKAGES = $(shell go list ./... | grep -v /vendor/)
+test: .gofmt .govet .gotest
+
+.gofmt:
+	OUT=$$(go fmt $(PACKAGES)); if test -n "$${OUT}"; then echo "$${OUT}" && exit 1; fi
```

Comment:
I tried the method you mentioned, but `gofmt` only supports absolute paths. My idea is to use `pwd` to get the absolute path of `image-tools`, but I don't know how to remove the `vendor` next .
Do you have any advice?